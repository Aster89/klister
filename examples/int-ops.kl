#lang "prelude.kl"

(example (+ 1 2))
(example (* 1 2))
(example (- 1 2))
(example (/ 10 3))
(define minus-two (negate 2))
(example (minus-two))
(example (abs minus-two))
(example (abs 4))
(example (negate 4))

(example (< 1 2))
(example (< 1 1))
(example (< 2 1))
(example (<= 1 2))
(example (<= 1 1))
(example (<= 2 1))
(example (> 1 2))
(example (> 1 1))
(example (> 2 1))
(example (>= 1 2))
(example (>= 1 1))
(example (>= 2 1))
(example (= 1 2))
(example (= 1 1))
(example (= 2 1))
(example (/= 1 2))
(example (/= 1 1))
(example (/= 2 1))
