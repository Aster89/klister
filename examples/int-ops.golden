3 : Integer
2 : Integer
-1 : Integer
3 : Integer
-2 : Integer
2 : Integer
4 : Integer
-4 : Integer
(true) : Bool
(false) : Bool
(false) : Bool
(true) : Bool
(true) : Bool
(false) : Bool
(false) : Bool
(false) : Bool
(true) : Bool
(false) : Bool
(true) : Bool
(true) : Bool
(false) : Bool
(true) : Bool
(false) : Bool
(true) : Bool
(false) : Bool
(true) : Bool
"4" : String
"-5" : String
